### ISSUE TITLE
TypeError: Cannot read property 'Line' of undefined

### ISSUE THREAD

**Issue by BuKinoshita** [paradox41/angular-chartist.js#71]
Why am I always getting this?

```jade
chartist(class="ct-chart ct-golden-section" chartist-data="chartData" chartist-chart-type="Line" chartist-chart-options="chartOptions" chartist-responsive-options="barResponsiveOptions")
```

```javascript
function dashboardController ($scope) {
    $scope.chartData = {
      labels: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'],
      series: [
          [0, 1, 2, 4, 7, 6, 9, 10, 8, 10, 14, 13, 16, 14, 17, 19, 20, 31, 32, 26, 36, 28, 31, 40, 26, 26, 43, 47, 55, 30],
          [0, 1, 2, 4, 4, 6, 6, 13, 9, 10, 16, 18, 21, 16, 16, 16, 31, 17, 27, 23, 31, 29, 35, 39, 30, 32, 26, 43, 51, 46],
          [0, 1, 3, 4, 6, 5, 11, 9, 11, 11, 13, 15, 14, 22, 20, 15, 31, 27, 25, 25, 36, 30, 37, 29, 29, 39, 40, 49, 34, 35],
          [0, 1, 3, 5, 7, 5, 9, 9, 10, 17, 13, 21, 14, 16, 23, 23, 25, 17, 24, 34, 27, 39, 33, 45, 47, 32, 40, 36, 49, 32],
          [0, 1, 3, 3, 7, 5, 8, 11, 12, 13, 16, 17, 20, 24, 27, 15, 22, 33, 35, 24, 32, 35, 41, 39, 24, 31, 51, 29, 45, 50]
      ]
    };

    $scope.chartOptions = {
      axisX: {
          labelInterpolationFnc: function(value) {
              return value;
          }
      }
    };
  }
```

**Comment by D-Bullock** [comment]
Hi, I'm getting the same issue. What was the resolution?

**Comment by paradox41** [comment]
Line might need to be in single quotes, like: chartist-chart-type="'Line'"
since it isn't being read from the scope - it's just a string.

**Comment by D-Bullock** [comment]
Wow that was the fastest reply I've ever had on Github. 
I think it might be due to the fact I don't use any moduling system, so in this line that determines the chart type fails
this.chart = Chartist[this.chartType](this.element, this.data, this.options, this.responsiveOptions);
as Chartist is null.

**Comment by paradox41** [comment]
Well Chartist has to be available on the window at least right? I use UMD
bundling, so it should work anywhere as long as libraries are included in
the correct order.

**Comment by D-Bullock** [comment]
Ah that was it, incorrect build order. Thanks for your help.

**Comment by b4dnewz** [comment]
Hi @willsoto, I don't want to open an issue since I'm not sure it is, but I'm having the same problem: `Chartist is undefined` but I'm loading the scripts in the correct order, using wiredep I enqueue first chartist than angular and angular-chartist (this is all verified with network inspector) but it still saying is undefined, instead inside any angular or js block if I log Chartist or window.Chartist is defined.. 

I've installed angular-chartist with this method described in the readme, with the last version.

> For example, with Bower (if not using bower-npm-resolver):
bower install 'https://github.com/willsoto/angular-chartist.js.git#<tag>' --save

I really can't find any explaination to this right now.. and It's a couple of hours I'm struggling..

Also I tried to debug this by placing this code:
```js
console.log('Angular', angular);
console.log('Chartist', Chartist);
console.log('Chartist', window.Chartist);
```
on the top of the dist file right after: `'use strict'`  and the strange thing is that `Chartist` __is undefined__ and `window.Chartist` __is defined__.

I hope this can help you to understand what's happening more than me, I think is something related to the UMD block but i'm not sure.

**Comment by willsoto** [comment]
What module loader are you using?

**Comment by b4dnewz** [comment]
the setup is that one from yeoman angular generator with grunt

On Sep 26, 2017 8:05 PM, "Will" <notifications@github.com> wrote:

What module loader are you using?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/willsoto/angular-chartist.js/issues/71#issuecomment-332284762>,
or mute the thread

**Comment by willsoto** [comment]
Could you paste some code snippets? Hard to tell what is going on without them

**Comment by b4dnewz** [comment]
It's a private large project but this are some snippets:

**bower.json**
```json
...
"overrides": {
    "angular-chartist.js": {
      "main": [
        "./dist/angular-chartist.js"
      ]
    }
  },
...
```

**app.js**
```js
angular
  .module('app', [
    ...
    'angular-chartist',
    ...
])
```

index.html
```html
<!-- build:js(.) scripts/vendor.js -->
<!-- bower:js -->
<script src="bower_components/chartist/dist/chartist.js"></script>
<script src="bower_components/angular/angular.js"></script>
...
<script src="bower_components/angular-chartist.js/dist/angular-chartist.js"></script>
<!-- endbower -->
<!-- endbuild -->
```
than I appended some console.log on your module: 
**angular-chartist.js**
```

**Comment by willsoto** [comment]
It's hard to really say what is going on. I don't know what kind of bundling Chartist uses, but I assume it is a UMD. UMD is a very well supported pattern and I don't write mine manually, [rollup takes care of that for me.](https://github.com/willsoto/angular-chartist.js/blob/master/rollup.config.js#L8-L15)

Nothing that you have pasted seems immediately wrong to me, but then again it has been awhile since I have used any components of that tool chain.

What if you try removing this line?

```js
Chartist = Chartist && 'default' in Chartist ? Chartist['default'] : Chartist;
```

**Comment by b4dnewz** [comment]
no, it's still not working the only one that has the value seems to be window object

**Comment by b4dnewz** [comment]
I've noticed another thing too, which is pretty strange to me, if I try to log this:
```js
console.log('Chartist', chartist);     // <-- reference error (error on all the page - code blocked )
console.log('Chartist', Chartist);  // <-- undefined (not reference error)
console.log('Chartist', window.Chartist); // <--- good
```
```
angular-chartist.js:8 Uncaught ReferenceError: chartist is not defined
    at angular-chartist.js:8
```
from this beheviour I can suppose that the Chartist variable was accidentally set to undefined otherwise it will throw a reference error like the other variables does.

**Comment by ayush** [comment]
@b4dnewz It looks like [this commit for correctly spell global name for Chartist](https://github.com/willsoto/angular-chartist.js/commit/66f0b930788e89c2d8e8b02e69227edecb1bc461) did not make it into the distribution. It still see the wrong spelling `Charitst ` when using 4.3.3 under `dist`. That causes `Chartist` references to become `undefined` 

Using the commit hash dependency does work however: `https://github.com/willsoto/angular-chartist.js.git#8e36314639873bd4ab64c7105b1fc8361960db40`

**Comment by willsoto** [comment]
Sorry about that. I just released v4.3.4 which should have the correct `dist`

**Comment by b4dnewz** [comment]
@ayush I didn't noticed that, many thanks for digging in the problem and many thanks @willsoto for fixing it
